policy-controller/k8s/index/src/outbound/index/unmeshed_network.rs

```diff
@@ -0,0 +1,230 @@
+use chrono::{offset::Utc, DateTime};
+use linkerd_policy_controller_k8s_api::policy::Cidr;
+use linkerd_policy_controller_k8s_api::{policy as linkerd_k8s_api, ResourceExt};
+use std::net::IpAddr;
+
+#[derive(Debug, Default)]
+pub(crate) struct Networks(pub Vec<Cidr>);
+
+#[derive(Debug, Default)]
+pub(crate) struct UnmeshedNetwork {
+    pub networks: Networks,
+    pub name: String,
+    pub namespace: String,
+    pub creation_timestamp: Option<DateTime<Utc>>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+struct MatchedUnmeshedNetwork {
+    matched_cidr: MatchedCidr,
+    name: String,
+    namespace: String,
+    creation_timestamp: Option<DateTime<Utc>>,
+}
+
+#[derive(Debug, PartialEq, Eq)]
+struct MatchedCidr(Cidr);
+
+// === impl MatchedUnmeshedNetwork ===
+
+impl Networks {
+    pub fn find_match(&self, ip: IpAddr) -> Option<Cidr> {
+        let ip: Cidr = ip.into();
+        self.0
+            .iter()
+            .filter(|c| c.contains(&ip))
+            .min_by(|a, b| a.size().cmp(&b.size()))
+            .cloned()
+    }
+}
+
+// === impl UnmeshedNetwork ===
+
+impl From<linkerd_k8s_api::UnmeshedNetwork> for UnmeshedNetwork {
+    fn from(u: linkerd_k8s_api::UnmeshedNetwork) -> Self {
+        let name = u.name_unchecked();
+        let namespace = u
+            .namespace()
+            .expect("UnmeshedNetwork must have a namespace");
+
+        UnmeshedNetwork {
+            name,
+            namespace,
+            networks: Networks(u.spec.networks.clone()),
+            creation_timestamp: u.creation_timestamp().map(|d| d.0),
+        }
+    }
+}
+
+// === impl MatchedUnmeshedNetwork ===
+
+impl From<(Cidr, &UnmeshedNetwork)> for MatchedUnmeshedNetwork {
+    fn from((cidr, unet): (Cidr, &UnmeshedNetwork)) -> Self {
+        Self {
+            name: unet.name.clone(),
+            namespace: unet.namespace.clone(),
+            matched_cidr: MatchedCidr(cidr),
+            creation_timestamp: unet.creation_timestamp,
+        }
+    }
+}
```

Comment:
this feels like a lurking footgun.

this is presuming that `find_match` was called to obtain this `Cidr`. maybe we could consider adding a `debug_assert!` to confirm (in test builds) this is never called with an incorrect cidr.

more ideally though, this doesn't seem like it should be an infallible `From<T>` if there are presumed invariants.

Reply:
good point, I think collapsing into one function would probably be best